app/models/task.rb

```diff
@@ -2,4 +2,17 @@
 
 class Task < ActiveRecord::Base
   validates :name, presence: true
+  validates :completed, on: :create, acceptance: { accept: false, message: 'Task cannot be marked as completed when created.' }
+
+  def self.get_list(completed_param)
```

Comment:
Yeah, I have added scope for <code>completed</code> and <code>incompleted</code>. And I still put the logic of checking task type in model layer so that controller layer will be very clean.